edgy/__init__.py

```diff
@@ -1,110 +1,177 @@
-from .core.db.fields.one_to_one_keys import OneToOne, OneToOneField
-from .core.db.models import Model, ModelRef, ReflectModel, StrictModel
-from .core.db.models.managers import Manager
-from .core.db.querysets import Prefetch, Q, QuerySet, and_, not_, or_
-from .core.extras import EdgyExtra
-from .core.signals import Signal
-from .core.utils.sync import run_sync
-from .exceptions import MultipleObjectsReturned, ObjectNotFound
+from edgy.core.db.querysets import Prefetch, Q, QuerySet, and_, not_, or_
+from edgy.core.utils.sync import run_sync
+
+if TYPE_CHECKING:
+    from .core.db.datastructures import Index, UniqueConstraint
+    from .core.signals import Signal
+    from .exceptions import MultipleObjectsReturned, ObjectNotFound
+
+
+class Instance(NamedTuple):
+    registry: Registry
+    app: Optional[Any] = None
+
 
 __all__ = [
+    "Instance",
+    "get_migration_prepared_registry",
+    "monkay",
     "and_",
     "not_",
     "or_",
     "Q",
+    "EdgyExtra",
+    "EdgySettings",
+    "files",
+    "Migrate",
+    "Prefetch",
+    "QuerySet",
+    "Signal",
+    "settings",
+    "run_sync",
+    # index and constraint
+    "Index",
+    "UniqueConstraint",
+    # some exceptions
+    "MultipleObjectsReturned",
+    "ObjectNotFound",
+    # constants
+    "CASCADE",
+    "RESTRICT",
+    "DO_NOTHING",
+    "SET_NULL",
+    "SET_DEFAULT",
+    "PROTECT",
+    "ConditionalRedirect",
+    # models
+    "ReflectModel",
+    "StrictModel",
+    "Model",
+    "ModelRef",
+    "Manager",
+    "RedirectManager",
+    # fields
+    "fields",
     "BigIntegerField",
     "BinaryField",
     "BooleanField",
-    "CASCADE",
-    "ConditionalRedirect",
     "CharField",
     "ChoiceField",
     "ComputedField",
     "CompositeField",
-    "Database",
-    "DatabaseURL",
     "DateField",
     "DateTimeField",
     "DurationField",
     "DecimalField",
-    "EdgyExtra",
-    "EdgySettings",
     "EmailField",
     "ExcludeField",
-    "fields",
-    "files",
     "FloatField",
+    "ImageField",
     "FileField",
     "ForeignKey",
-    "Index",
-    "IntegerField",
-    "JSONField",
-    "RefForeignKey",
-    "Manager",
-    "ManyToMany",
-    "ManyToManyField",
-    "Migrate",
-    "Model",
-    "ModelRef",
-    "MultipleObjectsReturned",
-    "ObjectNotFound",
     "OneToOne",
     "OneToOneField",
     "PasswordField",
     "PlaceholderField",
-    "Prefetch",
-    "QuerySet",
-    "ReflectModel",
-    "StrictModel",
-    "RESTRICT",
-    "Registry",
-    "SET_NULL",
-    "Signal",
     "SmallIntegerField",
     "TextField",
     "TimeField",
     "URLField",
     "UUIDField",
-    "UniqueConstraint",
-    "settings",
-    "run_sync",
+    "ManyToMany",
+    "ManyToManyField",
+    "IntegerField",
+    "JSONField",
+    "RefForeignKey",
+    "IPAddressField",
+    # base connection
+    "Database",
+    "DatabaseURL",
+    "Registry",
 ]
+
+monkay: Monkay[Instance, EdgySettings] = Monkay(
```

Comment:
lets put this in the conf package and import it here

Reply:
Monkay needs the globals of `__init__`, so it can inject the hooks.

Of course we could wrap it in a function...

Reply:
switching is possible via a partial/function but we still need to pass the globals dict

Reply:
can we move this topic to a later timepoint? This is just cosmetic.

Reply:
https://github.com/dymmond/edgy/issues/231

Reply:
Sure. Later then